Deploy initial 2022111400 version of plugin

rnathuji opened the pull request:
Summary of this plugin / deployment request:

* Description: This plugin is intended to encapsulate Moodle customizations that support research capabilities for the RAISE project. It is distinct from the [moodle-local_raisecli](https://github.com/openstax/moodle-local_raise) plugin in that the latter is specifically scoped to backend web service endpoints that are used to enable automation / batch jobs for RAISE (e.g. bulk creation of courses, etc.). 
* Initial release scope:
    * Create a `local_raise_user` database table which can be used to persist dynamically generated mappings between a Moodle user ID (incrementing `int`) and a UUID. These mappings are expected to be sparse relative to the native `user` table and created as needed when research related data is generated by non-Moodle front-end code.
    * Create a `local_raise_get_user` AJAX endpoint that can be used by research related front-end code to query a user UUID. The endpoint returns a raw UUID string value and UUID data encapsulated as a JWT so the latter can be used by custom frontend code when interacting with other services (e.g. an external research event sink for RAISE).
    * Define and integrate a plugin session cache that can be used to minimize database query / JWT generation overheads

marcghaly commented:
Hi @rnathuji,

Here is the outcome of my PR:

[N] Syntax

Extra spaces in `classes/user_helper.php` can be removed:
- line 71
- line 77
- line 78

in `classes/external/user.php`
- line 60
- line 61
- line 76 (unecessary line return)

If code is supposed to run on `PHP >= 5.4` (I assume it is), I would suggest changing `array()` calls to `[]`

[-] Output
[-] Component library

[N] Language

`$string['cachedef_userdata'] = 'RAISE user data';`

can be moved up to keep alphabetical order

[Y] Databases
[Y] Testing (instructions and automated tests)
[Y] Security
[Y] Privacy (see Privacy API)
[Y] Performance and Clustering
[N] Documentation

`get_or_create_jwt` param type should be `string` in classes/user_helper.php

`test_get_user_service` is missing cover informations (`@cover::`) in `local/raise/tests/user_test.php`

[-] Git
[-] Third party code
[Y] Sanity check
[-] Icons
[-] The Moodle mobile app / web services
[-] Accessibility (Optional)

Please let me know if you have any question, 

Regards,

Marc

rnathuji commented:
@marcghaly - I believe I've addressed all of your review comments. The one I was somewhat uncertain on was the last (I wasn't familiar with that annotation :sweat_smile: ), so let me know if [this change](https://github.com/openstax/moodle-local_raise/pull/7/commits/d18e05d18564cb812213355b013995a5f9abe8e9) addresses that okay.

marcghaly commented:
Hi @rnathuji,

All seems good to me, concerning shorthand `[]`, this can be added everywhere in the code instead of `array()` not only in `classes/user_helper.php`, this is not a blocker at all, I will let you merge changes, please advise me when done and I will deploy changes.

Ps: I am updating ticket

Regards,

Marc
